ResearchKit/Common/ORKAnswerFormat.m

```diff
@@ -1287,6 +1332,17 @@ - (NSString *)localizedInvalidValueStringWithAnswerString:(NSString *)text {
     return string;
 }
 
+- (NSString*)stringForAnswer:(id)answer {
+    if ([self isAnswerValid:answer]) {
+        NSNumberFormatter *formatter = [self makeNumberFormatter];
+        [formatter setPositiveSuffix: [NSString stringWithFormat:@" %@", self.unit]];
+        [formatter setNegativeSuffix:[NSString stringWithFormat:@"%@ ", self.unit]];
```

Comment:
Can you explain why there are two type of styles for unit string? And shall we consider when unit is nil or zero length?